### PR TITLE
Feature insect typ classifications

### PR DESCRIPTION
**Pull request by ChristianSalz**
# Add Insect Classification Feature

## Overview
This PR adds an optional binary classification layer on top of the existing YOLO detection model, enabling the system to distinguish between Musca domestica and other insects.

### New Files
- `TrainModelForInsectClassification.py` - Script to train a Keras CNN binary classifier
- `predict_Insect_Species.py` - Combined YOLO detection + classification inference script
- `InsectClassificationModel/` - Directory for storing trained classification models

### Updated Files
- `README.md` - Added section 9 documenting the classification feature
- `requirements.txt` - Added TensorFlow and related dependencies

### Binary Classification Model
- Custom CNN architecture with dropout for regularization
- Early stopping to prevent overfitting
- Trained on cropped insect images (Musca domestica vs Others)
- Achieves approximately 90% validation accuracy with 200+ images per class

### Combined Inference Pipeline
The new workflow operates as follows:
1. YOLO detects all insects in an image
2. Each detection is cropped and resized to 224x224 pixels
3. Classification model predicts House-Fly vs Others
4. Results are displayed with color-coded bounding boxes (red for Musca domestica, green for Others)
5. Summary statistics show counts per class

### Additional Improvements
- Support for both JPG and PNG images with automatic RGBA to RGB conversion
- Unique filename generation using UUID to prevent file overwrites when saving crops
- Confidence percentages displayed for each classification
- Detailed console output with per-insect classification results

###Usage
Check Md

Bildschirmfoto 2026-03-17 um 13 22 11

### CODE REVIEW

**Comment by ChristianSalz** [comment]
@lenntt yes i think this was / is much needed for NWO.

**Review by lenntt — COMMENTED** [review]
LGTM - as discussed in person, I think the biggest thing to consider is to make sure you want to classify to a specific species of (house)flies (where you've now annotated some flies incorrectly)

File: predict_Insect_Species.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+# Initialize counters
+class_counts = {
+    'Muscidae': 0,
+    'Others': 0
+}
+
+# Try to load a font (fallback to default if not available)
+try:
+    font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 20)
```
Comment (Bonajo) [inline predict_Insect_Species.py]: Linux doesn't get fancy font? 😒

**Review by Bonajo — APPROVED** [review]
